pkg/controller/cincinnati/new.go

```diff
@@ -333,6 +334,31 @@ func (k *kubeResources) newDeployment(instance *cv1beta1.Cincinnati) *appsv1.Dep
 	return dep
 }
 
+func (k *kubeResources) addPullSecret(instance *cv1beta1.Cincinnati) {
+	mode := int32(420) // 0644
+
+	// Add the volumeMount to the graph builder container
+	k.graphBuilderContainer.VolumeMounts = append(k.graphBuilderContainer.VolumeMounts, corev1.VolumeMount{
+			Name:      NamePullSecret,
+			ReadOnly:  true,
+			MountPath: "/var/lib/cincinnati/registry-credentials",
+	})
+
+	// Rebuild the deployment since we modified the container definition
+	k.deployment = k.newDeployment(instance)
```

Comment:
Rather than re-create this, it would be better to find a way to know up-front, before `newKubeResources()` gets called, whether this volume mount should be added. It looks like you followed the pattern from ensuring the configmap exists, and I think the same applies there. I'm thinking the `Reconcile` function should:

1. ensure the configmap exists in the namespace as appropriate
2. ensure the secret exists in the namespace as appropriate
3. call `newKubeResources` and let it add volume mounts for the configmap and secret if they exist
4. carry on with the rest of the "ensure" functions

That way we don't have to worry about re-creating resources, tracking whether anything else has already done anything with it in the meantime, etc.

WDYT? cc @djzager